### ISSUE TITLE
False negative for RS1024

### ISSUE THREAD

**Issue by bjornhellander** [dotnet/roslyn-analyzers#7472]
### Analyzer

**Diagnostic ID**: RS1024

### Analyzer source

**NuGet Package**: Microsoft.CodeAnalysis.Analyzers

**Version**: 3.11.0

### Describe the bug

RS1024 is not triggered, but I think it should.

### Steps To Reproduce

```
void M(List<INamedTypeSymbol> list, ITypeSymbol x, INamedTypeSymbol y)
{
    _ = list.Contains(x); // RS1024
    _ = list.Contains(y); // No RS1024
}
```
### Expected behavior

RS1024 should trigger on both calls to Contains.

### Actual behavior

It only triggers on the first.
